ci-operator/step-registry/assisted/baremetal/test/assisted-baremetal-test-commands.sh

```diff
@@ -26,32 +25,53 @@ trap collect_artifacts EXIT TERM
 # Tests execution
 set +e
 
-echo "### Copying test-list file"
-scp "${SSHOPTS[@]}" "${SHARED_DIR}/test-list" "root@${IP}:/tmp/test-list"
-
 echo "### Running tests"
-timeout --kill-after 10m 120m ssh "${SSHOPTS[@]}" "root@${IP}" bash -s "${OPENSHIFT_TESTS_IMAGE}" << "EOF"
+timeout --kill-after 10m 120m ssh "${SSHOPTS[@]}" "root@${IP}" "bash -s \"${OPENSHIFT_TESTS_IMAGE}\" \"${TEST_TYPE:-"list"}\" \"${TEST_SUITE:-"openshift/conformance/parallel"}\" \"${TEST_LIST:-""}\" \"${TEST_PROVIDER:-"baremetal"}\" \"${TEST_SKIPS}\"" << "EOF"
     set -x
 
-    function get_test_list() {
-        podman run --network host --rm -i -e KUBECONFIG=/tmp/kubeconfig -v ${kubeconfig}:/tmp/kubeconfig $1 \
-            openshift-tests run "openshift/conformance/parallel" --dry-run | \
-            grep -Ff /tmp/test-list
+    openshift_tests_image="$1"
+    test_type="$2"
+    test_suite="$3"
+    test_list="$4"
+    test_provider="$5"
+    test_skips="$6"
+
+    function get_baremetal_test_list() {
+        podman run --network host --rm -i \
+            -e KUBECONFIG=/tmp/kubeconfig -v "${KUBECONFIG}:/tmp/kubeconfig" "$1" \
+            openshift-tests run "$2" \
+            --dry-run \
+            --provider "{\"type\": \"$3\"}"
```

Comment:
sinces you are using named variable we should use them - makes it more readable

```suggestion
    function get_baremetal_test_list() {
        podman run --network host --rm -i \
            -e KUBECONFIG=/tmp/kubeconfig -v "${KUBECONFIG}:/tmp/kubeconfig" "${openshift_tests_image}" \
            openshift-tests run "$test_type" \
            --dry-run \
            --provider "{\"type\": \"$test_suite\"}"
```